spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastInstanceFactory.java

```diff
@@ -61,7 +62,16 @@ public HazelcastInstanceFactory(Config config) {
 
 	private Config getConfig(Resource configLocation) throws IOException {
 		URL configUrl = configLocation.getURL();
-		Config config = new XmlConfigBuilder(configUrl).build();
+		String configFileName = configUrl.getFile();
+
+		Config config;
```

Comment:
Rather than having a field initialization, I'd create a private method that returns the `Config` instance.

Reply:
Changed